PostFreeSurfer/PostFreesurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,433 @@
+# This block finalizes the output of TXw images (re-create the _restore version) before atlas registration.
+ log_Msg "Creating one-step resampled version of {T1w,T2w}_acpc_dc outputs"
+
+ # T1w
+#Applies GFC to T1w and T2w
+if (( TemplateProcessing == 0 )); then 
+
+  log_Msg "Timepoint $Timepoint_long: applying gain field to T1w,T2w images in longitudinal acpc_dc template space"
+  OutputT1wImage=${T1w_dir_long}/${T1wImage}_acpc_dc
+  #${FSLDIR}/bin/fslmaths $T1w_dir_long/T1w_acpc_dc.nii.gz -div $BiasField_long -mas "$T1w_dir_long"/"$T1wImageBrainMask"_orig.nii.gz $T1w_long -odt float
+
+  fslmaths ${OutputT1wImage} -abs ${OutputT1wImage} -odt float  # Use -abs (rather than '-thr 0') to avoid introducing zeros
+  fslmaths ${OutputT1wImage} -div ${T1w_dir_long}/BiasField_acpc_dc ${OutputT1wImage}_restore
+  #fslmaths ${OutputT1wImage}_restore -mas "$T1w_dir_long"/"$T1wImageBrainMask".nii.gz ${OutputT1wImage}_restore_brain
+
+  if (( Use_T2w )); then
+    OutputT2wImage=${T1w_dir_long}/${T2wImage}_acpc_dc
+    fslmaths ${OutputT2wImage} -abs ${OutputT2wImage} -odt float  # Use -abs (rather than '-thr 0') to avoid introducing zeros
+    fslmaths ${OutputT2wImage} -div ${T1w_dir_long}/BiasField_acpc_dc ${OutputT2wImage}_restore
+    #fslmaths ${OutputT2wImage}_restore -mas "$T1w_dir_long"/"$T1wImageBrainMask".nii.gz ${OutputT2wImage}_restore_brain
+  fi
+
+else #make tempate images
+    log_Msg "Template $Template: creating average templates and masks from timepoints"
+    OutputBrainMask="${T1w_dir_template}/$T1wImageBrainMask"
+    OutputT1wImage_unrestore="${T1w_dir_template}/${T1wImage}_acpc_dc"
+    OutputT1wImage="${OutputT1wImage_unrestore}_restore"
+    
+    OutputT2wImage_unrestore="${T1w_dir_template}/${T2wImage}_acpc_dc"
+    OutputT2wImage="${OutputT2wImage_unrestore}_restore"
+
+    nTP=${#timepoints[@]}
+    tp=${timepoints[0]}
+
+    brainmask_cmd="fslmaths $StudyFolder/$tp.long.$Template/T1w/$T1wImageBrainMask_orig"
+    template_cmd="fslmaths $StudyFolder/$tp.long.$Template/T1w/${T1wImage}_acpc_dc_restore"
+    template_cmd_unrestore="fslmaths $StudyFolder/$tp.long.$Template/T1w/${T1wImage}_acpc_dc"
+    template_cmd_t2w="fslmaths $StudyFolder/$tp.long.$Template/T1w/${T2wImage}_acpc_dc_restore"
+    template_cmd_t2w_unrestore="fslmaths $StudyFolder/$tp.long.$Template/T1w/${T2wImage}_acpc_dc"
+
+    for (( i=1; i<nTP; i++)); do
+        tp=${timepoints[i]}        
+        brainmask_cmd+=" -max $StudyFolder/$tp.long.$Template/T1w/$T1wImageBrainMask_orig"
+        template_cmd+=" -add $StudyFolder/$tp.long.$Template/T1w/${T1wImage}_acpc_dc_restore"
+        template_cmd_unrestore+=" -add $StudyFolder/$tp.long.$Template/T1w/${T1wImage}_acpc_dc"
+        template_cmd_t2w+=" -add $StudyFolder/$tp.long.$Template/T1w/${T2wImage}_acpc_dc_restore"
+        template_cmd_t2w_unrestore+=" -add $StudyFolder/$tp.long.$Template/T1w/${T2wImage}_acpc_dc"
+    done    
+
+    brainmask_cmd+=" $OutputBrainMask -odt float"
```

Comment:
Do we end up using a consistent final brain mask across the longitudinal timepoints?  I think that would be a good idea.